proposals/user-scripts-api.md

```diff
@@ -157,11 +162,54 @@ As mentioned in requirement A, the user script world can communicate with differ
   - Scripts registered via [`scripting.registerContentScripts()`](https://developer.chrome.com/docs/extensions/reference/scripting/#method-registerContentScripts), following the order they were registered in. Updating a content script doesn't change its registration order.
   - Scripts registered via `userScripts.register()`, following the order they were registered in. Updating a user script doesn’t change its registration order.
 - User scripts are always persisted across sessions, since the opposite behavior would be uncommon. (We may explore providing an option to customize this in the future.)
+- Unlike regular content scripts, `matches` is allowed to be optional when `includeGlobs` is specified. A user script matches a document when its URL matches either `matches` or `includeGlobs`.
+
+### RegisteredUserScript validation
+
+The `RegisteredUserScript` type is shared by `userScripts.register()` and
+`userScripts.update()`. All fields except `id` are declared as optional, to
+allow `userScripts.update()` to update individual properties.
+
+The following fields are required for a valid `RegisteredUserScript`:
+
+- `js` must be a non-empty array.
+- At least one of `matches` or `includeGlobs` must be a non-empty array.
```

Comment:
As mentioned before, Chrome currently requires `matches`, but it should not be. When optional, "obviously" one of the two arrays should be non-empty, as mentioned before at https://issues.chromium.org/issues/41483539#comment16

Reply:
Agree that matches should be optional, and with the other comments. However, it's not as clear when reading it on the updated proposal. "The following fields are required for a valid `RegisteredUserScript`" are only applicable when registering the script, right? Because updating a script doesn't require to provide at least one of `matches` or `includeGlobs`, or `js`.

What about:

```
The `RegisteredUserScript` type is shared by `userScripts.register()` and
`userScripts.update()`. All fields except `id` are declared as optional, to
allow `userScripts.update()` to update individual properties.

#### Requirements per method:
`userScripts.register()`:
- `js` must be present and a non-empty array.
- At least one of `matches` or `includeGlobs` must be a non-empty array.
 
`userScripts.update()` 
- individual properties may be `null` or omitted to leave the value 
unchanged. 
- To clear an array, an empty array can be passed. 
- The resulting script must be validated to make sure that the updated
script remains a valid script before it replaces a previous script.

#### Example
...
```